File: C/122_Best_Time_To_Buy_And_Sell_Stock_II.c
int maxProfit(int* prices, int pricesSize) {
    
    int i = 0;
    int sol = 0;
    int min = 10001;

    while(i < pricesSize) {
        //see if we have found a new price to buy
        if(prices[i] > min) {
            //buy it it
            sol += prices[i] - min;
        }
        //reset minimum
        min = prices[i];
        i++;
    }
    return sol;
}

File: Python/122_Best_Time_To_Buy_And_Sell_Stock_II.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        
        i = 0
        min = 10001
        sol = 0
        while i < len(prices):
            # check if this is the minimum
            if prices[i] > min:
                #if it is we can make more profit
                sol += prices[i] - min
            #reset the minimum
            min = prices[i]
            #keep going
            i += 1
        #return max profit
        return sol
